packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -327,6 +327,189 @@ export interface CommonAutoScalingGroupProps {
+
+  /**
+   * Launch template to use.
+   */
+  readonly launchTemplate: ec2.ILaunchTemplate;
+
+  /**
+   * Launch template overrides.
+   *
+   * The maximum number of instance types that can be associated with an Auto Scaling group is 40.
+   *
+   * The maximum number of distinct launch templates you can define for an Auto Scaling group is 20.
+   *
+   * @default - Do not provide any overrides
+   */
+  readonly launchTemplateOverrides?: LaunchTemplateOverrides[];
+}
+
+/**
+ * Indicates how to allocate instance types to fulfill On-Demand capacity.
+ */
+export enum OnDemandAllocationStrategy {
+  /**
+   * This strategy uses the order of instance types in the LaunchTemplateOverrides to define the launch
+   * priority of each instance type. The first instance type in the array is prioritized higher than the
+   * last. If all your On-Demand capacity cannot be fulfilled using your highest priority instance, then
+   * the Auto Scaling group launches the remaining capacity using the second priority instance type, and
+   * so on.
+   */
+  PRIORITIZED = 'prioritized',
+}
+
+/**
+ * Indicates how to allocate instance types to fulfill Spot capacity.
+ */
+export enum SpotAllocationStrategy {
+  /**
+   * The Auto Scaling group launches instances using the Spot pools with the lowest price, and evenly
+   * allocates your instances across the number of Spot pools that you specify.
+   */
+  LOWEST_PRICE = 'lowest-price',
+
+  /**
+   * The Auto Scaling group launches instances using Spot pools that are optimally chosen based on the
+   * available Spot capacity.
+   *
+   * Recommended.
+   */
+  CAPACITY_OPTIMIZED = 'capacity-optimized',
+
+  /**
+   * When you use this strategy, you need to set the order of instance types in the list of launch template
+   * overrides from highest to lowest priority (from first to last in the list). Amazon EC2 Auto Scaling
+   * honors the instance type priorities on a best-effort basis but optimizes for capacity first.
+   */
+  CAPACITY_OPTIMIZED_PRIORITIZED = 'capacity-optimized-prioritized',
+}
+
+/**
+ * InstancesDistribution is a subproperty of MixedInstancesPolicy that describes an instances distribution
+ * for an Auto Scaling group. The instances distribution specifies the distribution of On-Demand Instances
+ * and Spot Instances, the maximum price to pay for Spot Instances, and how the Auto Scaling group allocates
+ * instance types to fulfill On-Demand and Spot capacities.
+ *
+ * For more information and example configurations, see Auto Scaling groups with multiple instance types
+ * and purchase options in the Amazon EC2 Auto Scaling User Guide:
+ *
+ * https://docs.aws.amazon.com/autoscaling/ec2/userguide/asg-purchase-options.html
+ */
+export interface InstancesDistribution {
+  /**
+   * Indicates how to allocate instance types to fulfill On-Demand capacity. The only valid value is prioritized,
+   * which is also the default value.
+   *
+   * @default - prioritized
```

Comment:
@comcalvi, do we need to drop dash, `-`, when commenting the default value, which is what I got from @kaizen3031593 during code review on [this PR](https://github.com/aws/aws-cdk/pull/19214/files)?

Reply:
I will follow that comment too

Reply:
From what I understand, if the default value is some specific value, it should be used without a dash. If `@default` only follows a pure documentation then a dash is needed otherwise the first word will be recognized as the value.

Reply:
yes, the `-` should be dropped